integrations/livepeer/components/upload-file.tsx

```diff
@@ -0,0 +1,123 @@
+'use client'
+
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { useCreateAsset } from '@livepeer/react'
+import { useRouter } from 'next/navigation'
+import { useDropzone } from 'react-dropzone'
+import { BiVideoPlus } from 'react-icons/bi'
+
+import { Progress } from '@/components/ui/progress'
+
+import { FormLivepeerApiKey } from './form-livepeer-api-key'
+import { useIsLivepeerApiKeySet } from '../hooks/use-livepeer-api-key'
+
+export function UploadFile() {
+  const route = useRouter()
+  const [video, setVideo] = useState<File | undefined>()
+  const {
+    mutate: createAsset,
+    data: asset,
+    status,
+    progress,
+    error,
+  } = useCreateAsset(
+    video
+      ? {
+          sources: [{ name: video.name, file: video }] as const,
+        }
+      : null
+  )
+  const isLivepeerApiKeySet = useIsLivepeerApiKeySet()
+
+  const onDrop = useCallback(async (acceptedFiles: File[]) => {
+    if (acceptedFiles && acceptedFiles.length > 0 && acceptedFiles?.[0]) {
+      setVideo(acceptedFiles[0])
+    }
+  }, [])
+
+  const { getRootProps, getInputProps } = useDropzone({
+    accept: {
+      'video/mp4': ['.mp4', '.MP4'],
+    },
+    maxFiles: 1,
+    onDrop,
+  })
+
+  const isLoading = useMemo(() => status === 'loading' || (asset?.[0] && asset[0].status?.phase !== 'ready'), [status, asset])
+
+  const totalProgress = useMemo(
+    () =>
+      progress?.[0].phase === 'uploading'
+        ? Math.round((progress?.[0]?.progress * 100) / 2)
+        : progress?.[0].phase === 'processing'
+        ? Math.round((progress?.[0].progress * 100) / 2 + 50)
+        : 0,
+    [progress]
+  )
+
+  const progressFormatted = useMemo(
+    () =>
+      progress?.[0].phase === 'failed'
+        ? 'Failed to process video.'
+        : progress?.[0].phase === 'waiting'
+        ? 'Waiting...'
+        : progress?.[0].phase === 'uploading'
+        ? `Uploading: ${totalProgress}%`
+        : progress?.[0].phase === 'processing'
+        ? `Processing: ${totalProgress}%`
+        : null,
+    [progress]
+  )
+
+  useEffect(() => {
+    if (asset?.[0]) {
+      route.push(`/integration/livepeer/vod/${asset[0].id}`)
+    }
+  }, [asset])
```

Comment:
the `useCreateAsset` will already trigger a rerender if the `asset` changes, so adding this useEffect will cause an unnecessary rerender. You can just remove it. Also the vod url was wrong
```suggestion
  if (asset?.[0]) {
    route.push(`/integration/livepeer/vod/${asset[0].id}`)
  }
```